test/CSModule.t.sol

```diff
@@ -2957,6 +3009,400 @@ contract CsmQueueOps is CSMCommon {
+    {
+        uint256 noId = createNodeOperator(0);
+
+        _assertPriorityQueueEmpty();
+        _enablePriorityQueue();
+
+        uploadMoreKeys(noId, 8);
+        csm.obtainDepositData(3, ""); // no.enqueuedCount == 5
+
+        {
+            vm.expectEmit(true, true, true, true, address(csm));
+            emit IQueueLib.BatchEnqueued(noId, 2);
+
+            vm.expectEmit(true, true, true, true, address(csm));
+            emit IQueueLib.BatchEnqueued(noId, 10);
+
+            uploadMoreKeys(noId, 12);
+        }
+
+        BatchInfo[] memory exp = new BatchInfo[](2);
+
+        // The batch was partially consumed by the obtainDepositData call.
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 5 });
+        exp[1] = BatchInfo({ nodeOperatorId: noId, count: 2 });
+        _assertQueueState(PRIORITY_QUEUE, exp);
+
+        exp = new BatchInfo[](1);
+
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 10 });
+        _assertQueueState(REGULAR_QUEUE, exp);
+    }
+
+    function test_enqueueToPriorityQueue_EnqueuedWithDepositedMoreThanMaxDeposits()
+        public
+    {
+        uint256 noId = createNodeOperator(0);
+
+        _assertPriorityQueueEmpty();
+        _enablePriorityQueue();
+
+        uploadMoreKeys(noId, 12);
+        csm.obtainDepositData(3, ""); // no.enqueuedCount == 9
+
+        {
+            vm.expectEmit(true, true, true, true, address(csm));
+            emit IQueueLib.BatchEnqueued(noId, 12);
+
+            uploadMoreKeys(noId, 12);
+        }
+
+        BatchInfo[] memory exp = new BatchInfo[](1);
+
+        // The batch was partially consumed by the obtainDepositData call.
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 7 });
+        _assertQueueState(PRIORITY_QUEUE, exp);
+
+        exp = new BatchInfo[](2);
+
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 2 });
+        exp[1] = BatchInfo({ nodeOperatorId: noId, count: 12 });
+        _assertQueueState(REGULAR_QUEUE, exp);
+    }
+
+    function test_migrateToPriorityQueue_EnqueuedLessThanMaxDeposits() public {
+        uint256 noId = createNodeOperator(0);
+        uploadMoreKeys(noId, 8);
+
+        _assertPriorityQueueEmpty();
+        _enablePriorityQueue();
+
+        {
+            vm.expectEmit(true, true, true, true, address(csm));
+            emit IQueueLib.BatchEnqueued(noId, 8);
+
+            vm.prank(nodeOperator);
+            csm.migrateToPriorityQueue(noId);
+        }
+
+        BatchInfo[] memory exp = new BatchInfo[](1);
+
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 8 });
+        _assertQueueState(PRIORITY_QUEUE, exp);
+
+        exp[0] = BatchInfo({ nodeOperatorId: noId, count: 8 });
+        _assertQueueState(REGULAR_QUEUE, exp);
+    }
+
+    function test_migrateToPriorityQueue_EnqueuedMoreThanMaxDeposits() public {
+        uint256 noId = createNodeOperator(0);
+        uploadMoreKeys(noId, 15);
+
+        _assertPriorityQueueEmpty();
+        _enablePriorityQueue();
```

Comment:
Worth adding params to make further assets more readable and not just with some consts

Reply:
In all tests for sure

Reply:
Done